scripts/core/ui/components/SelectUser.tsx

```diff
@@ -1,176 +1,96 @@
-
         return (
-            <SelectWithTemplate
-                key={this.props.deskId}
-                label={gettext('Select a user')}
-                inlineLabel={true}
-                labelHidden={true}
-                getItems={(searchString) => {
-                    this.abortController?.abort();
-                    this.abortController = new AbortController();
-
-                    let url = '/users';
-
-                    if (this.props.deskId != null && this.props.deskId != '') {
-                        url = `/desks/${this.props.deskId}/users`;
-                    }
-
-                    const urlParams = {max_results: 50};
-
-                    if (searchString != null && searchString.length > 0) {
-                        urlParams['where'] = getUserSearchMongoQuery(searchString);
-                    }
-
-                    // Wrapping into additional promise in order to avoid having to handle rejected promise
-                    // in `SelectWithTemplate` component. The component takes a generic promise
-                    // as an argument and not a fetch result so it wouldn't be good to handle
-                    // fetch-specific rejections there.
-                    return new Promise((resolve) => {
-                        httpRequestJsonLocal<IRestApiResponse<IUser>>({
-                            method: 'GET',
-                            path: url,
-                            urlParams,
-                            abortSignal: this.abortController.signal,
-                        }).then((res) => {
-                            resolve(res._items);
-                        }).catch((err) => {
-                            // If user types something in the filter input all unfinished requests will be aborted.
-                            // This is expected behaviour here and should not throw an error.
-                            if (err?.name !== 'AbortError') {
-                                throw err;
-                            }
-                        });
-                    });
-                }}
-                value={this.state.selectedUser}
-                onChange={(user) => {
-                    this.setState({selectedUser: user});
-                    this.props.onSelect(user);
-                }}
-                getLabel={(option) => option.display_name}
-                itemTemplate={itemTemplate}
-                valueTemplate={valueTemplate}
-                areEqual={(a, b) => a._id === b._id}
-                autoFocus={this.props.autoFocus}
-                autoOpen={this.state.selectedUser == null}
-                width="100%"
-                zIndex={1050}
-                noResultsFoundMessage={gettext('No results found.')}
-                filterPlaceholder={gettext('Search...')}
-                data-test-id="select-user-dropdown"
-                required={!(this.props.clearable ?? true)}
-            />
+            options.length && (
```

Comment:
It is not allowed to return `undefined` from react components. There will always be 1 user - you - so you don't need the condition at all.